packages/edit-widgets/src/components/sidebar/widget-areas.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { blockDefault } from '@wordpress/icons';
+import { BlockIcon } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+export default function WidgetAreas( { selectedWidgetAreaId } ) {
+	const widgetAreas = useSelect(
+		( select ) => select( 'core/edit-widgets' ).getWidgetAreas(),
+		[]
+	);
+
+	const selectedWidgetArea =
+		selectedWidgetAreaId &&
+		widgetAreas?.find(
+			( widgetArea ) => widgetArea.id === selectedWidgetAreaId
+		);
```

Comment:
Nitpick: Not sure about the performance impact, but maybe we would want to `useMemo` this so that we don't have to recalculate `.find` every time this component renders.

Reply:
Done in 4b44c1b006!